include/fixed_containers/circular_integer_range_iterator.hpp

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include "fixed_containers/int_math.hpp"
+#include "fixed_containers/integer_range.hpp"
+#include "fixed_containers/iterator_utils.hpp"
+#include "fixed_containers/random_access_iterator.hpp"
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <tuple>
+
+namespace fixed_containers::circular_integer_detail
+{
+struct CyclesAndInteger
+{
+    std::int64_t cycles;
```

Comment:
For my understanding: What is the idea behind keeping track of the cycles?

Reply:
For FixedDeque, you can only really go `-1` or `+1` cycles, but `CircularIntegerRangeIterator` is more generic and handles arbitrary number of cycles.
The number of cycle is needed to differentiate between states that point to the same index. For `FixedDeque` specifically, it is used for differentiating between empty and full deque because those two states point to the same index but differ by 1 cycle.